Update sondes.yaml

FanHan-NOAA opened the pull request:
This PR updates sondes.yaml to use `ObsErrorModelHumidity@ObsFunction` for specific humidity error assignment. Compared to the original version, the new obs function **more accurately** mimics GSI error assignment for specific humidity, i.e., the RH error from `prepobs_errtable.txt` is scaled by **forecast saturation specific humidity** rather than **observed specific humidity**. 

For details about the obs function, check out https://github.com/JCSDA-internal/ufo/pull/1927 (UFO PR for the new obs function) and https://github.com/JCSDA-internal/jedi-docs/pull/426 (documentation for the new obs function)

The recently added GSI scheme of `SfcPCorrected` obs operator (https://github.com/JCSDA-internal/ufo/pull/1984) is also updated here.

FanHan-NOAA committed:
Update sondes.yaml

RussTreadon-NOAA reviewed (commented):
Change looks straight forward.   I wanted to test the updated `sondes.yaml` by using `ush/run_jedi_exe.py` to run `fv3jedi_var.x` with static B.   My yaml must have a problem.   `run_jedi_exe.py` fails when attempting to populate `bkg`.

RussTreadon-NOAA commented:
> Change looks straight forward. I wanted to test the updated `sondes.yaml` by using `ush/run_jedi_exe.py` to run `fv3jedi_var.x` with static B. My yaml must have a problem. `run_jedi_exe.py` fails when attempting to populate `bkg`.

The necessary `bkg` files exist for C768 but not C384 or C96.   Attempts to run C768 `3dhofx.yaml` or `3dvar.yaml` from PR #58 failed with exceptions.   Additional comments found in PR #58.

FanHan-NOAA commented:
@RussTreadon-NOAA Thanks a lot for testing. I think I may know the cause of the issue you described in #58. I will do some digging about the issue you found and hopefully get it resolved.

FanHan-NOAA committed:
Update sondes.yaml

FanHan-NOAA committed:
Update sondes.yaml

FanHan-NOAA committed:
change SfcPCorrected scheme to GSI

Commit:
Merge branch 'develop' into feature/update_sondes_qc

FanHan-NOAA commented:
I ran 3DVar experiment (c96) with original and updated `sondes.yaml` and compared the logs. The resulted increments are similar with largest differences in specific humidity (which is to be expected). The experiment with updated sondes.yaml resulted in a faster convergence with more norm reduction in less iteration, the final normal reduction is `4.7e-5` vs `5e-11` for original and updated, respectively. QC stats are also slightly different for specific humidity observations only.
